src/transformers/integrations/flex_attention.py

```diff
@@ -0,0 +1,44 @@
+from typing import Optional, Tuple
+
+import torch
+
+from ..utils import is_torch_greater_or_equal
+
+
+if is_torch_greater_or_equal("2.5"):
+    from torch.nn.attention.flex_attention import flex_attention
+
+
+def flex_attention_forward(
+    module: torch.nn.Module,
+    query: torch.Tensor,
+    key: torch.Tensor,
+    value: torch.Tensor,
+    attention_mask: Optional[torch.Tensor],
+    scaling: Optional[float] = None,
+    softcap: Optional[float] = None,
+    **kwargs,
```

Comment:
Head mask could be added as well as done in gpt neox.

Reply:
It's more like gpt_neo_x will get a flex attention function that has support for head mask! But good point!